### ISSUE TITLE
This was an error in Desktop & Screen Saver preferences

### ISSUE THREAD

**Issue by ckhk212** [JohnCoates/Aerial#537]
<img width="675" alt="screen shot 2018-10-13 at 3 52 38 pm" src="https://user-images.githubusercontent.com/561360/46909413-0d023c00-cf00-11e8-9f18-3550ef167cff.png">

Received an error message "This was an error in Desktop & Screen Saver preferences" with release 1.4 upon installation. Installed 1.4b without any problem.

MacBook Pro (13-inch, 2017, Four Thunderbolt 3 Ports) running 10.14

**Comment by glouel** [comment]
Can you clarify when you saw the error exactly ? Just after install ? Did it still pop when you closed System Preferences and opened it again ?

There seems to be a bug with System Preferences and Swift plugins not having all of their ressources reloaded, (that's what cause people to have the wrong version number showing after install). 

When you mean 1.4b, you mean 1.4b9 from this repo ? Please try reinstalling 1.4 on top, close System Preferences and reopen and let me know ? Thanks !

**Comment by fgaleano** [comment]
I get the exact same error after a fresh install, with the same steps as @ckhk212 in 10.14. I downloaded the latest release, 1.4.

**Comment by glouel** [comment]
@ckhk212 was this a fresh install or on top of 1.2 the first time ? 

@fgaleano you mention fresh install, you didn't have a previous version of Aerial prior to this that's correct ?

To both
- I assume you installed manually, not through brew ?
- are you behind a proxy/firewall
- which region did you set in MacOS settings
- would you be willing to go into your user library and try deleting  ~/Library/Preferences/ByHost/com.JohnCoates.Aerial.(UUID).plist

Sorry about that we'll try to find what happens.

**Comment by ckhk212** [comment]
@glouel. Please see below for my comments to your questions. Thanks.

- was this a fresh install or on top of 1.2 the first time ?

    Ya, fresh from scratch 10.14
- I assume you installed manually, not through brew ?

    Yup, just manually download release 1.4, unzip and double click the saver file
- are you behind a proxy/firewall

    Not that I am aware of
- which region did you set in MacOS settings

    Canada
- would you be willing to go into your user library and try deleting ~/Library/Preferences/ByHost/com.JohnCoates.Aerial.(UUID).plist

    Tried it, same problem. (deleted plist and removed the Aerial from system preferences and install again)
- long shot but could you possibly launch console.app, filter by "Aerial" top right, then try loading the screensaver again (go back to Destop & Screensaver to generate the error) and let me know of any lines you see in console ? If it doesn't show anything, you may want to try again without the filter and see if there's crash message from System Preferences

    Here you go "Aerial: Error can't load tvos11.json from cached directory"
- Do you know of anything special about your MacOS environment maybe (installed some specific dev tool, another bash, etc or something) ?

   No, quite a fresh install.
- Does ~/Library/Caches/Aerial exist ?

    Yup, with these files: `TVIdleScreenStrings.bundle`, `tvos10.json`, `entries.json`, `tvos11.json`, `resources.tar` and not seeing any `.mov` files

**Comment by glouel** [comment]
> @glouel. Please see below for my comments to your questions. Thanks.

First thanks for your patience in this and your answers. Going directly to it 👍 

> * long shot but could you possibly launch console.app, filter by "Aerial" top right, then try loading the screensaver again (go back to Destop & Screensaver to generate the error) and let me know of any lines you see in console ? If it doesn't show anything, you may want to try again without the filter and see if there's crash message from System Preferences
>   Here you go "Aerial: Error can't load tvos11.json from cached directory"

Ok that line helps a lot !

> * Does ~/Library/Caches/Aerial exist ?
>   Yup, with these files: `TVIdleScreenStrings.bundle`, `tvos10.json`, `entries.json`, `tvos11.json`, `resources.tar` and not seeing any `.mov` files

So you have the file but it can't be loaded. 

Can you tell me which size the tvos11.json file has on your system ? It should be 7 KB. If it's not, can you tell me the size and past the content ? I would guess it's an empty file ?

That's the file in question : https://sylvan.apple.com/Aerials/2x/entries.json

If it's an empty file, can you try deleting it and launching again ? See if the file gets downloaded correctly this time ?

**Comment by ckhk212** [comment]
No problem @glouel. The `tvos11.json` file is not empty and it is exactly 7kb. The content looks the same as https://sylvan.apple.com/Aerials/2x/entries.json. Also, tried deleing it and it downloads back properly, but same error.

**Comment by glouel** [comment]
Damn ! If that's ok with you I'll make a special debug build to try and track that, I'll ping you back here when done.

**Comment by glouel** [comment]
@ckhk212 if you feel like it, can you try this test build : 
https://github.com/glouel/Aerial/releases/download/1.4.1test5/Aerial.saver.zip ?

I did add a couple of extra checks so it might work outright. But I'd still love the console output (same thing, filter by "Aerial", you should see many lines now) as if it's what I think it was, there's something more complex going on.

Thanks !

**Comment by ckhk212** [comment]
BINGO! @glouel It works right away.

Here are some of the console messages I see:
<img width="1245" alt="screen shot 2018-10-14 at 12 48 20 pm" src="https://user-images.githubusercontent.com/561360/46919590-71300900-cfaf-11e8-98e6-7dedede0926a.png">

**Comment by glouel** [comment]
Ok progress ! Can you filter by AerialML instead in console while opening the panel ?

Hopefully looking for something like this : 
```
2018-10-14 18:49:49.165438+0200 AerialApp[16647:5897897] AerialML: Manifest init
2018-10-14 18:49:49.165644+0200 AerialApp[16647:5897897] AerialML: tvos10.json manifest is cached
2018-10-14 18:49:49.165670+0200 AerialApp[16647:5897897] AerialML: 10 true
2018-10-14 18:49:49.165720+0200 AerialApp[16647:5897897] AerialML: tvos11.json manifest is cached
2018-10-14 18:49:49.165734+0200 AerialApp[16647:5897897] AerialML: 11 true
2018-10-14 18:49:49.165775+0200 AerialApp[16647:5897897] AerialML: entries.json manifest is cached
2018-10-14 18:49:49.165788+0200 AerialApp[16647:5897897] AerialML: 12 true
2018-10-14 18:49:49.165856+0200 AerialApp[16647:5897897] AerialML: manifests are saved in preferences
2018-10-14 18:49:49.165868+0200 AerialApp[16647:5897897] AerialML: Loading from plist
2018-10-14 18:49:49.165883+0200 AerialApp[16647:5897897] AerialML: LSM
2018-10-14 18:49:49.184769+0200 AerialApp[16647:5897897] AerialML: lsm12
2018-10-14 18:49:49.276171+0200 AerialApp[16647:5897897] AerialML: lsm11
2018-10-14 18:49:49.282302+0200 AerialApp[16647:5897897] AerialML: lsm10
2018-10-14 18:49:49.363015+0200 AerialApp[16647:5897897] AerialML: post json loading
2018-10-14 18:49:49.363223+0200 AerialApp[16647:5897897] 66 videos processed !
```

I'm particularly interested in how many videos were processed (last line) and the lsm12/lsm11/lsm10 lines too ;) 

The descriptions might look weird in that build, I'll put a better one, you caught me in the middle of implementing font management. 

Thanks again !

**Comment by ckhk212** [comment]
Initially I don't see any `AerialML` in the log until I waited the screensaver to kick in, then I see this
```
default	13:02:32.661131 -0400	ScreenSaverEngine	AerialML: Manifest init
default	13:02:32.661637 -0400	ScreenSaverEngine	AerialML: tvos10.json manifest is cached
default	13:02:32.661775 -0400	ScreenSaverEngine	AerialML: 10 true
default	13:02:32.662071 -0400	ScreenSaverEngine	AerialML: tvos11.json manifest is cached
default	13:02:32.662295 -0400	ScreenSaverEngine	AerialML: 11 true
default	13:02:32.662520 -0400	ScreenSaverEngine	AerialML: entries.json manifest is cached
default	13:02:32.662652 -0400	ScreenSaverEngine	AerialML: 12 true
default	13:02:32.662980 -0400	ScreenSaverEngine	AerialML: manifests are NOT saved in preferences
default	13:02:32.663123 -0400	ScreenSaverEngine	AerialML: Not available from plist
default	13:02:32.663294 -0400	ScreenSaverEngine	AerialML: tvos10.json manifest is cached
default	13:02:32.663518 -0400	ScreenSaverEngine	AerialML: tvos11.json manifest is cached
default	13:02:32.663686 -0400	ScreenSaverEngine	AerialML: entries.json manifest is cached
default	13:02:32.663831 -0400	ScreenSaverEngine	AerialML: Manifests are cached on disk, loading
default	13:02:32.664259 -0400	ScreenSaverEngine	AerialML: 12path : 
default	13:02:32.665079 -0400	ScreenSaverEngine	AerialML: 11path : 
default	13:02:32.665488 -0400	ScreenSaverEngine	AerialML: 10path : 
default	13:02:32.666165 -0400	ScreenSaverEngine	AerialML: LSM
default	13:02:32.666331 -0400	ScreenSaverEngine	AerialML: lsm12
default	13:02:32.680808 -0400	ScreenSaverEngine	AerialML: lsm10
default	13:02:32.699955 -0400	ScreenSaverEngine	AerialML: post json loading
```
I don't see how many videos were processed though.

**Comment by glouel** [comment]
You may not see them depending on if the preview starts or not. So launching the saver will work too (you can set a hot corner if needed), I guess opening Aerial's preferences should do the trick too. 

No "XX videos processed !" at the end (after the "post json loading") ? 

Somehow it doesn't want to load your tvos11.json file. The path thing is intriguing, I'll test something here and report back.

**Comment by ckhk212** [comment]
> No "XX videos processed !" at the end (after the "post json loading") ?

No

When screensaver starts again, this is what the log shows
```
default	13:15:39.239287 -0400	com.apple.preference.desktopscreeneffect.remoteservice	AerialML: post json loading
default	13:15:51.532988 -0400	ScreenSaverEngine	AerialML: Manifest init
```

**Comment by glouel** [comment]
Sorry I edited my message while you were answering, see the edit above.

**Comment by ckhk212** [comment]
>Just to be clear, you see the UI when you click Screen Saver options right and it roughly works ?
With a non empty list ? :

Yup, it works and I query string "videos processed !" and I see `65 videos processed !`

**Comment by glouel** [comment]
Ok that confirms it. You're missing a couple of 4K videos, and I have no idea why ! I'll keep looking into it, thanks for your feedback!

**Comment by ckhk212** [comment]
No worry @glouel Thanks for the good work.

If you don't need more info, I will downgrade back to 1.4b9 and look forward seeing your next release. Looks sick.

**Comment by glouel** [comment]
Oh I think I got it, you probably have an APFS System Volume with case set to SENSITIVE instead of INSENSITIVE (something that highlighted a typo I made, oops !) ! 

Hold on 2 seconds before downgrading, I'll put a new build for you to test if that's ok.

**Comment by ckhk212** [comment]
AArrr.. right.. ! Sure, I can wait.

**Comment by glouel** [comment]
@ckhk212 
https://github.com/glouel/Aerial/releases/download/1.4.1beta1rc1/Aerial.saver.zip

If you can do the console thing again, and tell me if there's 66 videos this time.

**Comment by ckhk212** [comment]
@glouel do you mind double checking the link? no response on my end..

**Comment by glouel** [comment]
Let's try again : 

https://github.com/glouel/Aerial/releases/download/1.4.1beta1rc1/Aerial.saver.zip
Or the build page if that's better : https://github.com/glouel/Aerial/releases/tag/1.4.1beta1rc1

**Comment by ckhk212** [comment]
`default	15:15:56.321838 -0400	com.apple.preference.desktopscreeneffect.remoteservice	AerialML: 66 videos processed !`

Looking good

**Comment by glouel** [comment]
Closing as fixed with the release of 1.4.1
